src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -64,7 +64,7 @@
     /**
      * A submission can have a result and therefore, results are persisted and removed with a submission.
      */
-    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY)
+    @OneToOne(mappedBy = "submission", fetch = FetchType.LAZY, cascade = CascadeType.REMOVE, orphanRemoval = true)
```

Comment:
Please revert this change. We intentionally remove the annotations here, because they can lead to problems.

Reply:
I added it because I was not able to delete the submission. I googled for the SQL error message and found on stackoverflow that this is caused by a missing cascade delete. As the comment says the same I thought this is correct...

Reply:
You simply need to delete the result of the submission (manually) first, if it exists. Something like this should work

```
if (submission.getResult() != null) {
    resultRepository.delete(submission.getResult())
}
submissionRepository.delete(submission)
```

I think we can even extract a method for this into the SubmissionService, e.g. `deleteSubmission(Submission submission)`

We only need to make sure that the result is loaded eagerly or that the proxy object is accessed in the same transaction in which the submission is loaded.

Reply:
I implemented it in the SubmissionResource.java